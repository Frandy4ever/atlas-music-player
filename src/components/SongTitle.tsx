import { useState, useEffect } from "react";

interface SongTitleProps {
  songId: string;
}

const SongTitle: React.FC<SongTitleProps> = ({ songId }) => {
  const [title, setTitle] = useState<string>("");
  const [artist, setArtist] = useState<string>("");

  useEffect(() => {
    async function fetchSongDetails() {
      try {
        const response = await fetch(`/api/v1/songs/${songId}`);
        const data = await response.json();
        setTitle(data.title);
        setArtist(data.artist);
      } catch (error) {
        console.error("Error fetching song details:", error);
      }
    }

    fetchSongDetails();
  }, [songId]);

  return (
    <div className="text-center">
      <h2 className="text-xl font-bold text-slate-950 dark:text-slate-100">
        {title || "Unknown Title"}
      </h2>
      <p className="text-sm text-slate-600 dark:text-slate-400">
        {artist || "Unknown Artist"}
      </p>
    </div>
  );
};

export default SongTitle;
