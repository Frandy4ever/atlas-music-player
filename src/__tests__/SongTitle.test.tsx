import { render } from "@testing-library/react";
import SongTitle from "../components/SongTitle";

describe("SongTitle Component", () => {
  test("matches snapshot with song details", () => {
    const { asFragment } = render(<SongTitle songId="1" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
