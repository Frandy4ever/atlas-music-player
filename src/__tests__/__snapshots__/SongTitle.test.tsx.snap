// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SongTitle Component > matches snapshot with song details 1`] = `
<DocumentFragment>
  <div
    class="text-center"
  >
    <h2
      class="text-xl font-bold text-slate-950 dark:text-slate-100"
    >
      Unknown Title
    </h2>
    <p
      class="text-sm text-slate-600 dark:text-slate-400"
    >
      Unknown Artist
    </p>
  </div>
</DocumentFragment>
`;
